hazelcast/src/test/java/com/hazelcast/test/starter/GreaterLessPredicateConstructor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test.starter;
+
+import java.lang.reflect.Constructor;
+
+import static com.hazelcast.test.starter.HazelcastProxyFactory.proxyArgumentsIfNeeded;
+import static com.hazelcast.test.starter.ReflectionUtils.getFieldValueReflectively;
+
+public class GreaterLessPredicateConstructor extends AbstractStarterObjectConstructor {
```

Comment:
Could you introduce a smoke test for this class? This constructor class uses reflection so it would be good to have a fail-fast mechanism to identify incompatible changes in `GreaterLessPredicate` class. Maybe an `equals()` method override introduced in `GreaterLessPredicate` would help here too.